### PR TITLE
Update to Dagger 2.15

### LINKED ISSUE #591: Update to Dagger 2.15
## Feature Request
Update **Dagger** to version **_2.15_**
#### Feature Description
**Dagger** is a great dependency injection framework but presently, we are using **Dagger 2.0.2** which was released around 2 years ago. Many improvements have been made in 2 years so we must upgrade to the latest version.
After upgrading the version number, many changes might have to be made in the code since **Dagger** has changed a bit. I am quite versed in using this framework. So I can refactor the code.

@mhutti1 Please review this issue.

### LINKED ISSUE #615: Fix memory leaks
## Bug Report
We must fix all the memory leaks present in the app as it leads to **ANR** and crashes especially on older devices.

### Environment
* version of the software you use: 2.3
* device / operating system: Redmi Note 3 / Android 6.0.1
  
### The Bug:
Memory leaks are present in the following classes:
* KiwixWebView - The **Handler** class must **be** static.
* DownloadService - **DownloadFragment** field must **not** be static.
* KiwixDatabase - **KiwixDatabase** field must **not** be static.
* ZimContentProvider - **Context** field must **not** be static.
* ZimFileSelectFragment - **ZimManageActivity** must **not** be static.

@kelson42 @mhutti1 Please review this issue.

### PR DESCRIPTION
**Pull request by abdulwd**
Fixes #591 and part of #615 

Changes: Update **Dagger** version, fix some memory leaks, remove lint warning and some code cleanup.

@mhutti1 Please review. _networkTest_ and _downloadTest_ are sometimes failing which are also failing on _master_ branch. I request you to please merge this PR ASAP as afterward resolving conflicts will be difficult. :)
After it is merged I will fix #615.

### CODE REVIEW

**Comment by abdulwd** [comment]
@mhutti1 We should have **@PerFragment** scope for injecting fragments but the code is not at all modular. It has many hardcoded dependencies like we're using **LibraryFragment** in **KiwixMobileActivity** which must not be the case. After some more code refactoring in future we'll be able to use **Dagger** to inject fragments. By making the code loosely coupled it will be easier to write unit tests.

File: app/src/main/java/org/kiwix/kiwixmobile/ZimContentProvider.java [hunk 1]
```diff
@@ -70,29 +71,28 @@
...
   }
 
 
-  private static String getFulltextIndexPath(String file){
+  private static String getFulltextIndexPath(String file) {
     String[] names = {file, file};
 
     /* File might be a ZIM chunk like foobar.zimaa */
-    if (!names[0].substring(names[0].length() - 3).equals("zim")){
+    if (!names[0].substring(names[0].length() - 3).equals("zim")) {
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/ZimContentProvider.java]: I know these changes are correct however could they be moved to another commit/not done here. This is already a mammoth task to review.

File: app/src/main/java/org/kiwix/kiwixmobile/KiwixApplication.java [hunk 2]
```diff
@@ -25,38 +26,53 @@
...
     AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
   }
 
-  private ApplicationComponent applicationComponent;
+  @Inject
+  DispatchingAndroidInjector<Activity> mActivityInjector;
 
   public static KiwixApplication getInstance() {
-    return application;
+    return sApplication;
```
Comment (rashiq) [inline app/src/main/java/org/kiwix/kiwixmobile/KiwixApplication.java]: Please change it back to `application`. We don't follow hungarian notation.

File: app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java [hunk 3]
```diff
@@ -216,51 +216,74 @@
 
   private BookmarksDao bookmarksDao;
 
-  @BindView(R.id.toolbar) Toolbar toolbar;
+  @BindView(R.id.toolbar)
```
Comment (rashiq) [inline app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java]: You've submitted way too much - please use git patch mode to only submit actual changes, and no changes caused by you using a different auto format config.

File: app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java [hunk 4]
```diff
@@ -600,11 +631,11 @@ private void goToRateApp() {
     }
   }
 
-  private void updateTitle(String zimFileTitle) {
+  private void updateTitle(String zimFileTitle, ActionBar actionBar) {
```
Comment (rashiq) [inline app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java]: I can't seem to figure out what this change has to do with dagger. Please only commit code that has to do with the PR at hand.

**Comment by abdulwd** [comment]
@mhutti1 @rashiq I have tried to keep the changes as small as possible. I request you to merge this ASAP to avoid merge conflicts.

**Comment by abdulwd** [comment]
@mhutti1 Please merge this ASAP. I have removed the merge conflicts and all the tests pass locally. After it is merged I can move onto #98 and #615.
